### PR TITLE
Refactored modules to index.js to avoid duplicating names in require calls

### PR DESCRIPTION
**Pull request by PaulRBerg**
I like to go through repos and propose project-wide changes to get myself familiar with all the components. Feel free to avoid merging this if you think it's unnecessary.

Changelog:

- Updated the "main" in `package.json` to `index.js` so that you can have a global AZTEC object when linking the project globally via `npm ln`
- Added linting scripts
- Renamed modules to `index.js`, e.g. `bn128/bn128.js` is now `bn128/index.js`

I also updated all the require calls in tests and in all other places so that nothing got broken.

### CODE REVIEW

File: note/index.js [hunk 1]
```diff
@@ -2,9 +2,9 @@ const BN = require('bn.js');
 const web3Utils = require('web3-utils');
 const crypto = require('crypto');
 
-const secp256k1 = require('../secp256k1/secp256k1');
-const bn128 = require('../bn128/bn128');
-const setup = require('../setup/setup');
+const secp256k1 = require('../secp256k1/index');
```
Comment (ArnSch) [inline note/index.js]: Why the extra index?

File: eip712/sign.js [hunk 2]
```diff
@@ -7,7 +7,7 @@
  */
 
 const ecdsa = require('../secp256k1/ecdsa');
-const eip712 = require('./eip712');
+const eip712 = require('.');
```
Comment (ArnSch) [inline eip712/sign.js]: Would importing `./index` be clearer?

File: secp256k1/ecdsa.js [hunk 3]
```diff
@@ -9,7 +9,7 @@ const BN = require('bn.js');
 const web3Utils = require('web3-utils');
 const Web3EthAccounts = require('web3-eth-accounts');
 
-const secp256k1 = require('./secp256k1');
+const secp256k1 = require('./');
```
Comment (ArnSch) [inline secp256k1/ecdsa.js]: Would importing `./index` be clearer?

**Comment by ArnSch** [comment]
Looks good to go from me apart from the 3 super minor comments

**Comment by PaulRBerg** [comment]
Fixed all of them!

**Pull request merged**